### PR TITLE
Core: Confine URL parameter interactions to browser-specific code

### PR DESCRIPTION
**Pull request by gibson042**
Fixes gh-955

### CODE REVIEW

File: reporter/html.js [hunk 1]
```diff
@@ -113,8 +115,8 @@ function addClass( elem, name ) {
 	}
 }
 
-function toggleClass( elem, name ) {
-	if ( hasClass( elem, name ) ) {
+function toggleClass( elem, name, force ) {
+	if ( typeof force !== "undefined" ? !force : hasClass( elem, name ) ) {
```
Comment (leobalter) [inline reporter/html.js]: this is hard to read

File: reporter/urlparams.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	return;
+}
+
+var urlParams = getUrlParams();
+
+QUnit.urlParams = urlParams;
+
+// Match module/test by inclusion in an array
+QUnit.config.moduleId = [].concat( urlParams.moduleId || [] );
+QUnit.config.testId = [].concat( urlParams.testId || [] );
```
Comment (gibson042) [inline reporter/urlparams.js]: This leaves in place the current sloppiness (privileged parameters are applied immediately and unconditionally, while `urlConfig` parameters wait for runStart and only overwrite `undefined` values), but I think it's fine to save that cleanup for another ticket and commit (cf. https://github.com/jquery/qunit/issues/955#issuecomment-198474631 ).

**Comment by gibson042** [comment]
Any thoughts around how to test this?

File: reporter/urlparams.js (hunk 2 above)
Comment (leobalter) [inline reporter/urlparams.js]: yes. I'm +1 to progressive cleanups rather than holding PRs for too long.

**Comment by leobalter** [comment]
I really like what this PR does, and I'm afraid this might require defining several extra html files to observe the results of using urlParams.

**Comment by gibson042** [comment]
> I'm afraid this might require defining several extra html files to observe the results of using urlParams.

I don't think so... the new file really doesn't have that many responsibilities, and I'm pretty sure they can be tested together:
* parsing of a query string into a `QUnit.urlParams` object
* setting a few predefined properties on `QUnit.config`
* setting user-defined `QUnit.urlConfig` properties on `QUnit.config`

**Comment by leobalter** [comment]
Makes sense.

We already have tests for config when it's set, we only need to check if urlParams affect the config, right?

**Comment by leobalter** [comment]
where config my be QUnit.config, urlConfig, etc

File: reporter/urlparams.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+( function() {
```
Comment (trentmwillis) [inline reporter/urlparams.js]: Once #950 is merged, we can probably drop this closure.

**Comment by trentmwillis** [comment]
Overall, huge :+1: on this. As for the tests, what @gibson042 lists makes sense to me, so long as "setting a few predefined properties on `QUnit.config`" means "setting a few predefined properties on `QUnit.config` **via urlParams**".

Looking forward to the tests.

**Comment by gibson042** [comment]
Tests added! :tada:

**Comment by trentmwillis** [comment]
Tests look good to me. After a rebase I think this will be ready to go.

**Pull request closed without merging**